tests/unittests/Caffe2ImporterTest.cpp

```diff
@@ -1003,55 +970,27 @@ TEST(caffe2, dotProduct2D) {
   std::string NetWeightFilename(
       GLOW_DATA_PATH "tests/models/caffe2Models/empty_init_net.pbtxt");
 
-  Context ctx;
   Placeholder *output;
 
   // Input tensors.
-  const size_t kRows = 10;
-  const size_t kCols = 20;
-  Tensor X(ElemKind::FloatTy, {kRows, kCols});
-  Tensor Y(ElemKind::FloatTy, {kRows, kCols});
-  auto XH = X.getHandle();
-  auto YH = Y.getHandle();
+  constexpr std::size_t kRows = 10;
+  constexpr std::size_t kCols = 20;
+  auto type = mod.uniqueType(ElemKind::FloatTy, {kRows, kCols});
 
-  // Fill inputs with random values.
-  XH.randomize(-3.0, 3.0, mod.getPRNG());
-  YH.randomize(-3.0, 3.0, mod.getPRNG());
-
-  // Compute expected output.
-  Tensor O(ElemKind::FloatTy, {kRows});
-  auto OH = O.getHandle();
-
-  for (size_t i = 0; i < kRows; ++i) {
-    auto dotProduct = 0.0f;
-
-    // Compute dot product of the i-th row of X and Y.
-    for (size_t j = 0; j < kCols; ++j) {
-      dotProduct += (XH.at({i, j}) * YH.at({i, j}));
-    }
-
-    OH.at({i}) = dotProduct;
-  }
-
-  // Destroy the loader after the graph is loaded since the following execution
-  // will not depend on anyting from the loader.
+  // Destroy the loader after the graph is loaded, since it's not used in
```

Comment:
note, the real reason for early destruction is that we want to make sure that function `F` does not depend on anything stored inside the `Caffe2ModelLoader`